rclcpp/test/rclcpp/executors/test_executor_notify_waitable.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <memory>
+#include <stdexcept>
+
+#include "rclcpp/executors/single_threaded_executor.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "rcpputils/scope_exit.hpp"
+
+#include "rclcpp/executors/executor_notify_waitable.hpp"
+
+#include "../../utils/rclcpp_gtest_macros.hpp"
+
+
+class TestExecutorNotifyWaitable : public ::testing::Test
+{
+public:
+  void SetUp()
+  {
+    rclcpp::init(0, nullptr);
+  }
+
+  void TearDown()
+  {
+    rclcpp::shutdown();
+  }
+};
+
+TEST_F(TestExecutorNotifyWaitable, construct_destruct) {
+  {
+    auto waitable = std::make_shared<rclcpp::executors::ExecutorNotifyWaitable>();
+    waitable.reset();
+  }
+  {
+    auto on_execute_callback = []() {};
```

Comment:
we should add a test with a custom callback and verify that it's called during "execute"

Reply:
Still working on tests once I get my other PR rebased on here.